src/SAMLTrace.js

```diff
@@ -347,6 +373,12 @@ SAMLTrace.RequestItem = function(request) {
   if (this.request.saml != null || this.request.samlart != null) {
     this.availableTabs.push('SAML');
   }
+  if (this.request.samlresponse == true ) {
+    this.availableTabs.push('Response');
+  } 
+  if (this.request.samlrequest == true) {
+    this.availableTabs.push('Request');
+  }
```

Comment:
These lines could be simplified to:

    if (this.request.saml != null || this.request.samlart != null) {
      this.availableTabs.push('SAML');
      this.availableTabs.push('Summary');
    }

You could use the existing check for the presence of a SAML-token and push a "combined" summary-tab alongside with the SAML-tab.
By doing so the changes starting at line 283 and ending at line 323 wouldn't be necessary.